messenger/working-with-doctrine.rst

```diff
@@ -0,0 +1,64 @@
+.. index::
+    single: Messenger; Working with Doctrine
+
+Working with Doctrine
+=====================
+
+If your message handlers writes to a database it is a good idea to wrap all those
+writes in a single Doctrine transaction. This make sure that if one of your database
+query fails, then all queries are rolled back and give you a chance to handle the
+exception knowing that your database was not changed by your message handler(s).
+
+Next thing you need to do is to add the middleware to your bus configuration.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            # ...
+            buses:
+                messenger.bus.command:
+                    middleware:
+                        - validation
```

Comment:
are we purposely also including validation?

Reply:
That's a bit unrelated with the content of this doc indeed, but a good practice, as content of the command in a CQRS application are most of the input by users (on contrary of events which are mostly crafted by developers from already validated data).